app/services/parameters/caster.rb

```diff
@@ -0,0 +1,125 @@
+module Foreman
+  module Parameters
+    class Caster
+      attr_reader :value
+
+      def initialize(item, options = {})
+        defaults = {
+          :attribute_name => :value,
+          :to => :string
+        }
+        options.reverse_merge!(defaults)
+        @item, @options = item, options
+        @value = @options[:value] || @item.send(@options[:attribute_name])
+      end
+
+      def cast!
+        @item.send("#{@options[:attribute_name]}=", casted_value)
+      end
+
+      def cast
+        casted_value
+      end
+
+      private
+
+      def casted_value
+        case @options[:to]
+        when :string
+          cast_string
+        when :integer
+          cast_integer
+        when :real
+          cast_real
+        when :boolean
+          cast_boolean
+        when :array
+          cast_array
+        when :hash
+          cast_hash
+        when :json
+          cast_json
+        when :yaml
+          cast_yaml
+        else
+          Rails.logger.warn("Unable to type cast #{value} to #{@options[:to]}")
+          if @options[:raise_on_failed_cast]
```

Comment:
Where exactly?  Do you mean at https://github.com/theforeman/foreman/pull/2399/files#diff-bec20964ead6a00adfe25f42324af3e1L153 ?  In fact, that code seems to expect a TypeError.